### PR TITLE
[#4146] Apply soap timeout for Stuf-ZDS client

### LINKED ISSUE #4146: Default timeout is always used for SOAP services
### Product versie / Product version

2.6.2

### Omschrijf het probleem / Describe the bug

When setting a custom timeout for a soapservice this is not reflected in the real request, timeout will still occur after 10s and this is also showing in the logs:
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='httpstat.us', port=443): Read timed out. (read **timeout=10.0**)
image

This seems to only affect the SOAP services, as other services use the timeout from the configuration:
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='httpstat.us', port=443): Read timed out. (read **timeout=11**)
image

### PR DESCRIPTION
**Pull request by Viicos**
Fixes #4146

### CODE REVIEW

File: src/stuf/stuf_zds/tests/test_client.py [hunk 1]
```diff
@@ -155,3 +156,13 @@ def test_non_latin1_characters(self):
...
+
+    @tag("gh-4146")
+    def test_custom_timeout_applied(self):
+        stuf_service = StufServiceFactory.build(
+            soap_service__url="https://httpstat.us/", soap_service__timeout=1
+        )
+        client = StufZDSClient(stuf_service, {})  # type: ignore
+
+        with self.assertRaises(Timeout):
+            client.get("https://httpstat.us/200?sleep=3000")
```
Comment (vaszig) [inline src/stuf/stuf_zds/tests/test_client.py:168]: Can't we test this with a mocked request and not a real one?

**Review by vaszig — COMMENTED** [review]
I am not sure about this fix (if this is the only solution), maybe we should wait for @sergei-maertens 's opinion as well.

File: src/stuf/stuf_zds/client.py [hunk 2]
```diff
@@ -114,13 +114,15 @@ def StufZDSClient(service: StufService, options: ZaakOptions) -> "Client":
...
-    return Client.configure_from(
+    client = Client.configure_from(
         factory,
         options=options,
         failure_log_callback=partial(logevent.stuf_zds_failure_response, service),
         success_log_callback=partial(logevent.stuf_zds_success_response, service),
         **init_kwargs,
     )
+    client.request = partial(client.request, timeout=service.soap_service.timeout)
+    return client
```
Comment (sergei-maertens) [inline src/stuf/stuf_zds/client.py:125]: ape-pie supports specifying `timeout` at the "client" level: https://github.com/maykinmedia/ape-pie/blob/979972db0797e0c0b50aa14db801c7e8e9ac170f/tests/test_config_adapter.py#L56

The timeout kwarg should be set in the `get_client_init_kwargs` helper.

File: src/stuf/stuf_zds/tests/test_client.py (hunk 1 above)
Comment (sergei-maertens) [reply src/stuf/stuf_zds/tests/test_client.py:168]: yes, see the test in ape-pie: https://github.com/maykinmedia/ape-pie/blob/979972db0797e0c0b50aa14db801c7e8e9ac170f/tests/test_config_adapter.py#L56

You must always be able to run the test suite offline.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/stuf/stuf_zds/tests/test_client.py (hunk 1 above)
Comment (Viicos) [reply src/stuf/stuf_zds/tests/test_client.py:168]: I got my inspiration from this test:
https://github.com/open-formulieren/open-forms/blob/264e0644f88efe5734889677700cf86df538b7fa/src/soap/tests/test_client.py#L246-L259

Probably it will need to be updated

Comment (sergei-maertens) [reply src/stuf/stuf_zds/tests/test_client.py]: I vaguely remember there being a very specific edge cases that required this approach due to the zeep abstractions

**Review by sergei-maertens — APPROVED** [review]

**Comment by sergei-maertens** [comment]
Backport: c79d5f9cbb7c97c8a3f651ca7db3fed2b859b3b9 (to 2.6.x only, older versions don't have this feature)
